packages/firestore/test/unit/api/database.test.ts

```diff
@@ -306,13 +306,13 @@ describe('Settings', () => {
     expect(db._getSettings().experimentalForceLongPolling).to.be.false;
   });
 
-  it('long polling should be disabled if force=false', () => {
+  it('long polling should be in auto-detect mode if force=false', () => {
     // Use a new instance of Firestore in order to configure settings.
     const db = newTestFirestore();
     db._setSettings({
       experimentalForceLongPolling: false
```

Comment:
by the looks of it, whether `experimentalForceLongPolling` is set to false or not does not change the settings below (as it looks the same from the test above).

However, setting `experimentalForceLongPolling` to false should actually disable auto detect long polling too, is that right?

Reply:
If both `experimentalForceLongPolling=true` and `experimentalAutoDetectLongPolling=true` then an exception will be thrown:

https://github.com/firebase/firebase-js-sdk/blob/e147219ee85d739615879dcfc2ea25d2b54c1ce3/packages/firestore/test/unit/api/database.test.ts#L269-L280

Those two settings are mutually exclusive. As a result, only one of `forceLongPolling` and `detectBufferingProxy` will be `true` in `WebChannelOptions`

https://github.com/firebase/firebase-js-sdk/blob/e147219ee85d739615879dcfc2ea25d2b54c1ce3/packages/firestore/src/platform/browser/webchannel_connection.ts#L202-L203

Does that address your concern?